src/D2L.CodeStyle.Analyzers/Threading/UseConfigureAwaitFalseAnalyzer.cs

```diff
@@ -0,0 +1,141 @@
+
+			context.RegisterCodeFix(
+				CodeAction.Create(s_title, c => AddConfigureAwait(context.Document, awaitExpression, c), s_title),
+				diagnostic);
+
+		}
+
+		private async Task<Document> AddConfigureAwait(
+			Document document,
+			AwaitExpressionSyntax awaitExpression,
+			CancellationToken ct
+		) {
+
+			var configureAwaitExpr = SyntaxFactory.MemberAccessExpression(
+				SyntaxKind.SimpleMemberAccessExpression,
+				awaitExpression.Expression,
+				SyntaxFactory.IdentifierName("ConfigureAwait"));
+
+			var invocExpression =
+				SyntaxFactory.InvocationExpression(configureAwaitExpr)
+							 .WithArgumentList(
+								 SyntaxFactory.ArgumentList(
+									 SyntaxFactory.SingletonSeparatedList<ArgumentSyntax>(
+										 SyntaxFactory.Argument(SyntaxFactory.LiteralExpression(SyntaxKind.FalseLiteralExpression)))));
+
+			var newAwait = SyntaxFactory.AwaitExpression(invocExpression);
+
+			var root = await document.GetSyntaxRootAsync(ct);
+			var newRoot = root.ReplaceNode(awaitExpression, newAwait);
+
+			var newDoc = document.WithSyntaxRoot(newRoot);
+
+			return newDoc;
+
+
+		}
+
+		public override FixAllProvider GetFixAllProvider() {
+			return WellKnownFixAllProviders.BatchFixer;
+		}
+
+		public override ImmutableArray<string> FixableDiagnosticIds
+		{
+			get { return ImmutableArray.Create<string>(UseConfigureAwaitFalseAnalyzer.DIAGNOSTIC_ID); }
+		}
+
+	}
+
+	[DiagnosticAnalyzer(LanguageNames.CSharp)]
+	public class UseConfigureAwaitFalseAnalyzer : DiagnosticAnalyzer {
+
+		public const string DIAGNOSTIC_ID = "D2LAwaitFalse";
+
+		// You can change these strings in the Resources.resx file. If you do not want your analyzer to be localize-able, you can use regular strings for Title and MessageFormat.
+		// See https://github.com/dotnet/roslyn/blob/master/docs/analyzers/Localizing%20Analyzers.md for more on localization
+		private static readonly LocalizableString s_title = "Awaitable's should specify a ConfigureAwait";
+		private static readonly LocalizableString s_description = "Awaitable's should use ConfigureAwait(false)";
+		private const string CATEGORY = "Naming";
+
+		private static readonly DiagnosticDescriptor s_rule = new DiagnosticDescriptor(DIAGNOSTIC_ID, s_title, s_description, CATEGORY, DiagnosticSeverity.Warning, isEnabledByDefault: true, description: s_description);
+
+
+		public override void Initialize(AnalysisContext context) {
+			context.RegisterSyntaxNodeAction(AnalyzeSymbol, SyntaxKind.AwaitExpression);
+		}
+
+		public override ImmutableArray<DiagnosticDescriptor> SupportedDiagnostics
+		{
+			get { return ImmutableArray.Create<DiagnosticDescriptor>(s_rule); }
+		}
+
+		private static void AnalyzeSymbol(SyntaxNodeAnalysisContext context) {
+
+			bool isConfigured = context.Node.DescendantNodes()
+					.OfType<MemberAccessExpressionSyntax>()
+					.Any(x => IsConfigureAwaitFunction(x, context.SemanticModel));
+
+			if (!isConfigured) {

```

Comment:
So I'm up in the air about it. If you specified ConfigureAwait(true), I kind of assume you know what you are doing and so I would hate to show an error in that case.